### DISCUSSION TITLE
Artificial satellites visibility

### DISCUSSION THREAD

**Discussion by WilliamDrap** [Stellarium/stellarium#2624] [Q&A]
Hello,
I try to set/unset the artificial satellites visibility with a script (or the web api). Can't find the correct syntax. Could you help me ?
thank you
William

**Comment by alex-w** [comment]
Is it not possible.

Please describe use cases for change the visibility of individual satellites.

**Reply by WilliamDrap** [reply]
we are working on an easy to use interface for children (or occasional stellarium users). we need to interact on the visibility of all satellites at the same time (on/off) 
William

**Reply by WilliamDrap** [reply]
Just these two parameters would be fine :) Thank you 
![2022-09-02_16h10_05](https://user-images.githubusercontent.com/96690830/188167892-85f5a7a0-3ec2-4392-bf20-df84869ddbd0.png)

**Reply by gzotti** [reply]
Just use the main switch on the bottom icon bar. Or its equivalent in scripting syntax:

https://stellarium.org/doc/0.22/group__satellites.html#abae2eff2c523c353e356ded8c50e3a4a

`Satellites.setFlagHintsVisible(bool)   
`

**Reply by WilliamDrap** [reply]
Yes it is the answver !  I used the following syntax

_Satellites.setFlagHintsVisible(true);
Satellites.setFlagIconicMode(true);
Satellites.setFlagLabelsVisible(true);_
